### DISCUSSION TITLE
Why Apply can't access to event's metadata ?

### DISCUSSION THREAD

**Discussion by nlips** [rasmus/EventFlow#878] [Q&A]
I use the AddUserClaimsMetadata configuration method to have the user's identity in all events metadata.
But how to use this data in the AggregateState apply function ?
My aggregate has a notion of owner that I wish to deduce from its creation event.

**Comment by rasmus** [comment]
Currently some event metadata is generated when the event is emitted, other when the event is saved into the event store.

**Comment by rasmus** [comment]
The Emit method does allow you to provide additional metadata. While it may not be exactly what you wanted, if you provide the metadata there, it would be the same when subscribers and read models read it
